supervisord.py

```diff
@@ -47,14 +49,27 @@ def check(self, instance):
             AgentCheck.UNKNOWN: 0
         }
 
-        # Report service checks and uptime for each process
-        proc_names = instance.get('proc_names')
-        if proc_names and len(proc_names):
-            processes = [server.supervisor.getProcessInfo(p) for p in proc_names]
-        else:
-            processes = server.supervisor.getAllProcessInfo()
+        processes = server.supervisor.getAllProcessInfo()
+        proc_regex = instance.get('proc_regex', [])
+        proc_names = instance.get('proc_names', [])
+        monitored_processes = []
+
+        if not len(proc_regex) and not len(proc_names):
+            monitored_processes = processes
+
+        if len(proc_regex):
+            for pattern, process in itertools.product(proc_regex, processes):
+                if re.match(pattern, process['name']):
```

Comment:
I slightly prefer `if re.match(...) and process not in monitored_processes:` to the nested if statements